opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/aggregation/dsl/MetricAggregationBuilder.java

```diff
@@ -86,8 +93,22 @@ public MetricAggregationBuilder(ExpressionSerializer serializer) {
       NamedAggregator node, Object context) {
     Expression expression = node.getArguments().get(0);
     Expression condition = node.getDelegated().condition();
+    Boolean distinct = node.getDelegated().distinct();
     String name = node.getName();
 
+    if (distinct) {
+      switch (node.getFunctionName().getFunctionName()) {
+        case "count":
+          return make(
```

Comment:
why the distinct count doesn't support filter? do we support count(distinct *)?

Reply:
Filter is added, but count(distinct *) is not supported since it does not follow the MySQL grammar. Instead, we could consider support distinct count with multi field, for example `count(distinct firstname, lastname, age, ...)`